elastic-agent-client.proto

```diff
@@ -2,25 +2,105 @@
 // or more contributor license agreements. Licensed under the Elastic License;
 // you may not use this file except in compliance with the Elastic License.
 
-
 syntax = "proto3";
-package elasticagentclient;
 
-message ConfigRequest {
-    string config = 1;
+package proto;
+
+option cc_enable_arenas = true;
+option go_package = "pkg/proto;proto";
+
+// A status observed message is streamed from the application to Elastic Agent.
+//
+// This message contains the currently applied `config_state_idx` (0 in the case of initial start, 1 is the first
+// applied config index) along with the status of the application. In the case that the sent `config_state_idx`
+// doesn't match the expected `config_state_idx` that Elastic Agent expects, the application is always marked as
+// `CONFIGURING`.
+message StateObserved {
+    // Status codes for the current state.
+    enum Status {
+        // Application is starting.
+        STARTING = 0;
+        // Application is currently configuring.
+        CONFIGURING = 1;
+        // Application is in healthy state.
+        HEALTHY = 2;
+        // Application is working but in a degraded state.
+        DEGRADED = 3;
+        // Application is failing completely.
+        FAILED = 4;
+        // Application is stopping.
+        STOPPING = 5;
+    }
+    // Token that is used to unique identify the application to agent. When agent started this
+    // application it would have provided it this token.
+    string token = 1;
+    // Current index of the applied configuration.
+    uint64 config_state_idx = 2;
+    // Status code.
+    Status status = 3;
+    // Message for the health status.
+    string message = 4;
 }
 
-message ConfigResponse {
+// A state expected message is streamed from the Elastic Agent to the application informing the application
+// what Elastic Agent expects the applications state to be.
+message StateExpected {
+    enum State {
+        // Expects that the application is running.
+        RUNNING = 0;
+        // Expects that the application is stopping.
+        STOPPING = 1;
```

Comment:
are we expecting it to be stopping or stopped, just a small detail

Reply:
Once stopped the application should be off and the GRPC should be disconnected at that point. So I was going with `stopping` which also goes with `running` both being `-ing` based.